### DISCUSSION TITLE
What happens when there are more then 1 ems-esp detected by HA?

### DISCUSSION THREAD

**Discussion by g6094199** [emsesp/EMS-ESP32#1195] [General]
Hi,

i have to ems-esp (with different names an different heating systems) in the network and try to connect them to 1 HA instance.
i can see both ems in the mosquitto broker and they update nicely but HA seems to get fucked up. i see a mixture of settings ob both ems in the HA mqtt plugin.

so is there something wrong.  has this ever been tested? is this a ems or HA problem? or both in combination?

sash

**Comment by proddy** [comment]
yeah, you're right. I just did a quick test. EMS-ESP will use the same unique id for the device in HA, so you'll end up with duplicates. I'll make a new issue and see how best to fix it. Maybe by using the hostname for the `ids` in
```json
  "dev": {
    "ids": [
      "ems-esp"
    ]
```
